docs/formatter.md

```diff
@@ -225,6 +225,81 @@ def f(x):
 [literal blocks]: https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#literal-blocks
 [`code-block` and `sourcecode` directives]: https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block
 
+## F-string formatting
+
+_Stabilized in 2025 style with Ruff 0.9.0_
+
+By default, Ruff formats f-string expressions. This is a [known deviation](formatter/black.md#f-strings)
+from Black, which does not format f-strings.
+
+Ruff employs several heuristics to determine how an f-string should be formatted, including the
+following considerations:
+
+* The quote style for both outer and nested f-strings
+* Whether to break the f-string into multiple lines
+* When to skip formatting the f-string altogether
+
+### Quotes
+
+Ruff will use the configured quote style for the f-string expression unless doing so would result in
+invalid syntax for the target Python version. In such cases, Ruff will avoid changing the
+quote style.
+
+For example, if a self-documenting f-string contains a string literal with the configured quote
+style, Ruff cannot use the same quote style for the f-string itself when targeting Python versions
```

Comment:
I think this is slightly misleading. Ruff can change the quotes for:

```py
f'{10 + len("hello \" ")}'
```

It gets formatted to 

```
f"{10 + len('hello " ')}"
```

It can only not change the quotes if they're inside a debug expression (or format specifier). Let's double check with the implementation if there are other reasons when the preferred quote style can't be applied

Reply:
Thanks for sharing that although I'm referring to a self-documenting f-string which shouldn't change the quote.

I'll add other reasons when we don't change the quotes as well.

Reply:
oh, I didn't notice that (and probably wouldn't have understood what it is). It might be worth adding a reference link